docs/v3tov4.md

```diff
@@ -0,0 +1,283 @@
+                                               # and also sent as inventory.
+    "name":"integration name",
+    "version":"integration version"
+  },
+  "data":[                                    # List of entities
+    {
+      "metrics":[                             # list of metrics using the dimensional metric format
+        {
+          "name":"redis.metric1",
+          "type":"count",                     # gauge, count, summary or pdelta
+          "value":93, 
+          "attributes":{}                     # set of key-value pairs that define the dimensions of the metric
+        }
+      ],
+      "entity":{                              # this object is optional
+        "name":"redis:192.168.100.200:1234",  # unique entity name per account.
+        "type":"RedisInstance",               # entity's category
+        "displayName":"my redis instance",    # human readable name
+        "tags":{}                             # key-value pairs that will be also added as attributes to all metrics 
+                                              # and events
+      },
+      "inventory":{...},                      # Inventory data format has not changed
+      "events":[...]                          # Events format has not changed
+    }
+  ]
+}
+```
+
+## GoSDK v4 API changes
+
+This section enumerates the main changes you have to keep in mind if you want to upgrade from GoSDK v3.x to v4.x
+
+### Entities
+
+In previous versions of the SDK, we had a distinction between Local and Remote entities. You could get the 
+`Local Entity` calling the method
+
+`func (i *Integration) LocalEntity() *Entity`
+
+or create a `Remote Entity` calling
+
+`func (i *Integration) Entity(name, namespace string, idAttributes ...IDAttribute) (e *Entity, err error)`
+
+In SDK v4, there's only one way of creating entities:
+
+`func (i *Integration) Entity(name, type, displayName string) (e *Entity, err error)`
+
+* `name` must be an unique value per customer account because it uniquely identifies your Entity. It cannot be empty and 
+  the SDK cannot validate nor enforce that uniqueness, so it's up to the client to define a naming schema that produces 
+  unique names.
+
+  For example, using the following schema `<service_name>:<ip|hostname>:<port>`, we could create these names: 
+  `redis:192.168.100.200:12345` or `mysql:my_server_hostname:8086`
+
+  For the ip or the hostname, please avoid using the values `127.0.0.1` or `localhost` as they don't produce unique 
+  identifiers.
+
+* `type` describes the Entity's category. It cannot be empty. Some examples: "RedisInstance", "DockerContainer", 
+  "K8SPod", "CassandraCluster", "MySQLInstance", "KafkaBroker", "NginxServer", etc.
+
+* `displayName` is a friendly human readable name to be used in the UI. It can be empty.
+
+#### Migrating from LocalEntity
+
+If you are using the LocalEntity, you need to use the new Entity method with the appropriate parameters.
```

Comment:
I was thinking that you would not provide an entity, in this case? Perhaps I'm missing something.